pkg/fmap/fmap.go

```diff
@@ -127,36 +143,57 @@ func Read(f io.Reader) (*FMap, *Metadata, error) {
 		return nil, nil, err
 	}
 
-	// Check for too many fmaps.
-	if bytes.Count(data, Signature) >= 2 {
-		return nil, nil, errors.New("Found multiple signatures")
-	}
+	// Loop over __FMAP__ occurrences until a valid header is found
+	start := 0
+	ValidFmaps := 0
+	var fmap FMap
+	var fmapMetadata Metadata
+	for {
+		if start >= len(data) {
+			break
+		}
 
-	// Check for too few fmaps.
-	start := bytes.Index(data, Signature)
-	if start == -1 {
-		return nil, nil, errors.New("Cannot find fmap signature")
-	}
+		next := bytes.Index(data[start:], Signature)
+		if next == -1 {
+			break
+		}
+		start += next
+
+		if start >= len(data) {
+			break
+		}
 
-	// Reader anchored to the start of the fmap
-	r := bytes.NewReader(data[start:])
+		// Reader anchored to the start of the fmap
+		r := bytes.NewReader(data[start:])
 
-	// Read fields.
-	var fmap FMap
-	if err := readField(r, &fmap.Header); err != nil {
-		return nil, nil, err
-	}
-	fmap.Areas = make([]Area, fmap.NAreas)
-	if err := readField(r, &fmap.Areas); err != nil {
-		return nil, nil, err
+		// Read fields.
+		var TestFmap FMap
+		if err := readField(r, &TestFmap.Header); err != nil {
+			return nil, nil, err
+		}
+		if headerValid(&TestFmap.Header) {
+			fmap = TestFmap
+			ValidFmaps++
+		} else {
+			start += len(Signature)
+			continue
+		}
+		fmap.Areas = make([]Area, fmap.NAreas)
+		if err := readField(r, &fmap.Areas); err != nil {
+			return nil, nil, err
+		}
+		// Return useful metadata
+		fmapMetadata = Metadata{
+			Start: uint64(start),
+		}
+		start += len(Signature)
 	}
-
-	// Return useful metadata
-	fmapMetadata := Metadata{
-		Start: uint64(start),
+	if ValidFmaps >= 2 {
+		return nil, nil, errors.New("Found multiple fmap")
```

Comment:
```suggestion
		return nil, nil, errors.New("Found multiple fmaps")
```

Reply:
Additionally, do not capitalize: https://github.com/golang/go/wiki/CodeReviewComments#error-strings